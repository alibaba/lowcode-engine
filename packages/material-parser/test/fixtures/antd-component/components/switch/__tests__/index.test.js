import React from 'react';
import { mount } from 'enzyme';
import Switch from '..';
import focusTest from '../../../tests/shared/focusTest';
import { resetWarned } from '../../_util/warning';
import mountTest from '../../../tests/shared/mountTest';
import rtlTest from '../../../tests/shared/rtlTest';

describe('Switch', () => {
  focusTest(Switch);
  mountTest(Switch);
  rtlTest(Switch);

  it('should has click wave effect', async () => {
    const wrapper = mount(<Switch />);
    wrapper.find('.ant-switch').getDOMNode().click();
    await new Promise(resolve => setTimeout(resolve, 0));
    expect(wrapper.render()).toMatchSnapshot();
  });

  it('warning if set `value`', () => {
    resetWarned();

    const errorSpy = jest.spyOn(console, 'error').mockImplementation(() => {});
    mount(<Switch value />);
    expect(errorSpy).toHaveBeenCalledWith(
      'Warning: [antd: Switch] `value` is not a valid prop, do you mean `checked`?',
    );
    errorSpy.mockRestore();
  });
});
