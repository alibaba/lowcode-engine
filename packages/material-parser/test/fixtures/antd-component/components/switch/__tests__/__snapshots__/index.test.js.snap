// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch rtl render component should be rendered correctly in RTL direction 1`] = `
<button
  aria-checked="false"
  class="ant-switch-rtl ant-switch"
  role="switch"
  type="button"
>
  <span
    class="ant-switch-inner"
  />
</button>
`;

exports[`Switch should has click wave effect 1`] = `
<button
  ant-click-animating="true"
  aria-checked="false"
  class="ant-switch"
  role="switch"
  type="button"
>
  <span
    class="ant-switch-inner"
  />
  <div
    class="ant-click-animating-node"
    style="border-color: buttonface;"
  />
</button>
`;
